### PR TITLE
Fixes a crash that happens when using AL_usdmayaProxyShapeSelect with pseudo-root "/"

### PR DESCRIPTION
**Pull request by pmolodo**
## Description (this won't be part of the changelog)

NOTE: this branches off of https://github.com/AnimalLogic/AL_USDMaya/pull/142, so please check that PR first.

Currently, if you try to use the AL_usdmayaProxyShapeSelect command to select the pseudo-root ("/"), it will trigger a crash. (Though sometimes not immediately.)  This fixes this, so that trying to select the root will just select the ProxyShape object itself, instead.

### Fixed
- Fixes crash when selecting "/" with AL_usdmayaProxyShapeSelect; now selects proxy shape object

## Checklist (Please do not remove this line)
- [X] Make sure the Title and Description of the PR make sense and  provide sufficient context for your work
- [X] Do any added files have the correct AL Apache Licence Header?
- [X] Are there Doxygen comments in the headers?
- [X] Are any new features, behaviour changes documented in the .md format [documentation](https://github.com/AnimalLogic/AL_USDMaya/docs)?
- [X] Have you added, updated tests to cover new features and behaviour changes?
- [X] Have you filled out at least one changelog entry?

### CODE REVIEW

File: lib/AL_USDMaya/AL/usdmaya/nodes/ProxyShapeSelection.cpp [hunk 1]
```diff
@@ -1044,6 +1044,20 @@ bool ProxyShape::doSelect(SelectionUndoHelper& helper, const SdfPathVector& orde
   }
   MStringArray newlySelectedPaths;
 
+  auto removeFromMSel = [](MSelectionList& sel, const MObject& toRemove) -> bool {
+    for(uint32_t i = 0; i < sel.length(); ++i)
```
Comment (robthebloke) [inline lib/AL_USDMaya/AL/usdmaya/nodes/ProxyShapeSelection.cpp]: Take a copy of sel.length() - it's a dso exported method, so will always be re-evaluated.

File: lib/AL_USDMaya/AL/usdmaya/nodes/ProxyShapeSelection.cpp [hunk 2]
```diff
@@ -1078,6 +1092,14 @@ bool ProxyShape::doSelect(SelectionUndoHelper& helper, const SdfPathVector& orde
       uint32_t hasNodesToCreate = 0;
       for(auto prim : insertPrims)
       {
+        if(prim.IsPseudoRoot())
+        {
+          // For pseudo root, just modify maya's selection, don't alter
+          // our internal paths
+          newlySelectedPaths.append(MFnDagNode(thisMObject()).fullPathName());
```
Comment (robthebloke) [inline lib/AL_USDMaya/AL/usdmaya/nodes/ProxyShapeSelection.cpp]: This feels a little dirty. It doesn't handle instancing, not that I have a better suggestion right now...

**Review by robthebloke — COMMENTED** [review]
One tiny wee issue (sorry, I know I'm pedantic!). We may have to have a larger discussion wrt instancing support (which is on my todo list).

File: lib/AL_USDMaya/AL/usdmaya/nodes/ProxyShapeSelection.cpp (hunk 1 above)
Comment (elrond79) [reply lib/AL_USDMaya/AL/usdmaya/nodes/ProxyShapeSelection.cpp]: done!

File: lib/AL_USDMaya/AL/usdmaya/nodes/ProxyShapeSelection.cpp (hunk 2 above)
Comment (elrond79) [reply lib/AL_USDMaya/AL/usdmaya/nodes/ProxyShapeSelection.cpp]: Agreed, this made my skin itch too.  However, (as you also pointed out), it's far from the first place in the code base where we're getting a dag path from only an MObject - for instance, in the same file:

https://github.com/AnimalLogic/AL_USDMaya/blob/develop/lib/AL_USDMaya/AL/usdmaya/nodes/ProxyShapeSelection.cpp#L518

...and trying to get the plugin as a whole to behave well with instances seemed a much larger scope than this small fix.

However... I can also understand not wanting to make a problem worse.  Unfortunately, the only way I can think of to do this "right" would be to modify the signature of "doSelect" to take an additional argument - a dagPath (or at least an instance index... though I think a dagPath would be preferable) for the particular proxyShape instance we're selection.  Then, we'd also need to modify ProxyShapeCommandBase::getShapeNode() to return a dag path, not a ProxyShape*.

That seemed like a wider-reaching change than I wanted for this bugfix PR... but if you're on board with it, I could do that chunk, which would be a first small step towards instance compatibility...

**Review by robthebloke — APPROVED** [review]

**Pull request merged**